flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -2338,69 +2340,76 @@ class FMCMainDisplay extends BaseAirliners {
     }
 
     //Needs PR Merge #3082
-    trySetThrustReductionAccelerationAltitude(s) {
+    async trySetThrustReductionAccelerationAltitude(s) {
         if (s === FMCMainDisplay.clrValue) {
             this.thrustReductionAltitudeIsPilotEntered = false;
             this.accelerationAltitudeIsPilotEntered = false;
-            CDUPerformancePage.UpdateThrRedAccFromOrigin(this, true, true);
+            CDUPerformancePage.UpdateThrRedAccFromOrigin(this);
             return true;
         }
 
         const origin = this.flightPlanManager.getOrigin();
-        const elevation = origin ? origin.altitudeinFP : 0;
+
+        let elevation = SimVar.GetSimVarValue("GROUND ALTITUDE", "feet");
+        if (origin) {
+            elevation = await this.facilityLoader.GetAirportFieldElevation(origin.icao);
+        }
+
         const minimumAltitude = elevation + 400;
 
-        let newThrRedAlt = null;
-        let newAccAlt = null;
+        let checkThrRedAlt = this.thrustReductionAltitude;
+        let checkAccAlt = this.accelerationAltitude;
 
         let [thrRedAlt, accAlt] = s.split("/");
 
-        if (thrRedAlt && thrRedAlt.length > 0) {
-            if (!/^\d{3,5}$/.test(thrRedAlt)) {
+        if (thrRedAlt && thrRedAlt.length > 0 || accAlt && accAlt.length > 0) {
```

Comment:
I think this `if` block is a bit convoluted. I believe we can clean this up a bit by removing some redundant null checks on `thrRedAlt` and `accAlt` for example. Furthermore, it seems that this block does not check whether the altitudes are in the required range, and that this check is done further down. Are you sure this is intended?

Reply:
The reason this may look a bit overdone is because the values can be input as a pair  1200/1200 or as indivuals 1200 or /1200. It is just to check how the data is being input. Then the checks on the validity of the values is checked.

Reply:
Having thought about the inputs. It is possible that 1200/q200 could be entered. If the if block separated the two parts 1200 would be past as valid. When in fact the whole line should be returned as invalid format.

Reply:
> I think this `if` block is a bit convoluted. I believe we can clean this up a bit by removing some redundant null checks on `thrRedAlt` and `accAlt` for example. Furthermore, it seems that this block does not check whether the altitudes are in the required range, and that this check is done further down. Are you sure this is intended?

I am doing a lot more pilot entry validation. Also exiting the function earlier and checking if can exit with more often so the wait and heavy lifting code to update the perf page is only done when it is needed.

async trySetThrustReductionAccelerationAltitude(s) {
        if (s === FMCMainDisplay.clrValue) {
            this.thrustReductionAltitudeIsPilotEntered = false;
            this.accelerationAltitudeIsPilotEntered = false;
            CDUPerformancePage.UpdateThrRedAccFromOrigin(this);
            return true;
        }
        if (s.length == 0 || s.length > 11) {
            this.addNewMessage(NXSystemMessages.formatError);
            return false;
        }
        if (s.length > 6 && !/^\d{3,5}\/\d{3,5}$/.test(s)) {
            this.addNewMessage(NXSystemMessages.formatError);
            return false;
        }

        let [thrRedAlt, accAlt] = s.split("/");

        if (thrRedAlt === undefined || thrRedAlt == "") {
            thrRedAlt = 0;
        }

        if (accAlt === undefined || accAlt == "") {
            accAlt = 0;
        }

        if (isNaN(thrRedAlt) || isNaN(accAlt)) {
            this.addNewMessage(NXSystemMessages.formatError);
            return false;
        }

        if (thrRedAlt > 45000 || accAlt > 45000) {
            this.addNewMessage(NXSystemMessages.entryOutOfRange);
            return false;
        }

        if (thrRedAlt > 0 && !/^\d{3,5}$/.test(thrRedAlt)) {
            this.addNewMessage(NXSystemMessages.formatError);
            return false;
        }

        if (accAlt > 0 && !/^\d{3,5}$/.test(accAlt)) {
            this.addNewMessage(NXSystemMessages.formatError);
            return false;
        }

        let currentThrRedAlt = this.thrustReductionAltitude;
        let currentAccAlt = this.accelerationAltitude;
        let updateAccAlt = false;
        let updateThrRedAlt = false;

        const origin = this.flightPlanManager.getOrigin();
        let elevation = SimVar.GetSimVarValue("GROUND ALTITUDE", "feet");
        if (origin) {
            elevation = await this.facilityLoader.GetAirportFieldElevation(origin.icao);
        }

        const minimumAltitude = elevation + 400;

        if (thrRedAlt > 0) {
            thrRedAlt = parseInt(thrRedAlt);
            thrRedAlt = Math.round(thrRedAlt / 10) * 10;
            currentThrRedAlt = thrRedAlt;
        }

        if (accAlt > 0) {
            accAlt = parseInt(accAlt);
            accAlt = Math.round(accAlt / 10) * 10;
            currentAccAlt = accAlt;
        }

        if (thrRedAlt > 0 && (thrRedAlt < minimumAltitude || thrRedAlt > currentAccAlt)) {
            this.addNewMessage(NXSystemMessages.entryOutOfRange);
            return false;
        } else if (thrRedAlt > 0) {
            updateThrRedAlt = true;
        }

        if (accAlt > 0 && (accAlt < minimumAltitude || accAlt < currentThrRedAlt)) {
            this.addNewMessage(NXSystemMessages.entryOutOfRange);
            return false;
        } else if (accAlt > 0) {
            updateAccAlt = true;
        }

        if (updateThrRedAlt == true) {
            this.thrustReductionAltitude = thrRedAlt;
            this.thrustReductionAltitudeIsPilotEntered = true;
            SimVar.SetSimVarValue("L:AIRLINER_THR_RED_ALT", "Number", thrRedAlt);
        } else {
            this.thrustReductionAltitude = currentThrRedAlt;
            SimVar.SetSimVarValue("L:AIRLINER_THR_RED_ALT", "Number", currentThrRedAlt);
        }

        if (updateAccAlt == true) {
            this.accelerationAltitude = accAlt;
            this.accelerationAltitudeIsPilotEntered = true;
            SimVar.SetSimVarValue("L:AIRLINER_ACC_ALT", "Number", accAlt);
        } else {
            this.accelerationAltitude = currentAccAlt;
            SimVar.SetSimVarValue("L:AIRLINER_ACC_ALT", "Number", currentAccAlt);
        }
        return true;
    }

Reply:
PR updated to force pilot input to use 4 digits. eg if eng out acc is 900 then 0900 must be entered. This is IAW the manual.